ci/run_clang_tidy.sh

```diff
@@ -70,7 +70,7 @@ elif [[ "${BUILD_REASON}" != "PullRequest" ]]; then
 else
   echo "Running clang-tidy-diff against master branch..."
   git fetch https://github.com/envoyproxy/envoy.git master
-  git diff "${SYSTEM_PULLREQUEST_TARGETBRANCH:-refs/heads/master}..HEAD" | filter_excludes | \
+  git diff "$(git merge-base HEAD FETCH_HEAD)..HEAD" | filter_excludes | \
```

Comment:
The reason to use SYSTEM_PULLREQUEST_TARGETBRANCH is it will generate large diff for PR agains release branch and take forever.

Also, AZP run's on HEAD at GH pull request merge commit (i.e. refs/pull/9140/merge), so the merge-base will just be HEAD, so this generates no diff.

Reply:
I see, so the issue isn't with `SYSTEM_PULLREQUEST_TARGETBRANCH` but `HEAD`. Sounds like the correct fix here would be to replace `HEAD` with `SYSTEM_PULLREQUEST_SOURCEBRANCH` ?

Reply:
actually, I think just `git diff "${SYSTEM_PULLREQUEST_TARGETBRANCH}"` will work, without `..HEAD`?